src/server/req_delete.c

```diff
@@ -343,88 +383,54 @@ req_deletejob(struct batch_request *preq)
 		qdel_mail = 0;
 	else
 		qdel_mail = 1;
-	parent = chk_job_request(jid, preq, &jt);
-	if (parent == NULL)
+
+	if (FALSE == chk_job_request(jid, preq, &jt, &js, &pjob, &parent, &offset))
 		return; /* note, req_reject already called */
 
 	if (delhist) {
-		rc = check_deletehistoryjob(preq);
+		rc = check_deletehistoryjob(preq, pjob, parent, jt);
 		if (rc == TRUE)
 			return;
 	}
 
 
-	if (jt == IS_ARRAY_NO) {
+	if ((jt == IS_ARRAY_NO) || (jt == IS_ARRAY_Single)) {
 
-		/* just a regular job, pass it on down the line and be done
+		/* just a regular job or a subjob, pass it on down the line and be done
 		 * If the request if to purge the history of the job then set ji_deletehistory to 1
 		 */
-		if (delhist)
-			parent->ji_deletehistory = 1;
-		req_deletejob2(preq, parent);
-		return;
+		if (pjob != NULL) {
+			if (delhist)
+				pjob->ji_deletehistory = 1;
```

Comment:
This flag is only required for regular jobs since we do not support deleting history of jobs at the moment.

Reply:
Fixed by rvert in rc 20 https://github.com/PBSPro/pbspro/pull/590/commits/b01fcac3e94363966f635e6fad726a5342288a79